bundles/org.openhab.binding.enigma2/src/main/java/org/openhab/binding/enigma2/internal/Enigma2Client.java

```diff
@@ -0,0 +1,336 @@
+        if(asking) {
+            boolean state = getBoolean(document, "e2state");
+            if (state) {
+                String[] text = getString(document, "e2statetext").split(" ");
+                answer = text[text.length - 1].replace("!", "");
+                asking = false;
+                lastAnswerTime = LocalDateTime.now();
+            }
+        }
+    }
+
+    private void processVolumeResult(Document document) {
+        volume = getInt( document,"e2current");
+        mute = getBoolean(document, "e2ismuted");
+    }
+
+    private void processEpgResult(Document document) {
+        title = getString(document, "e2eventtitle");
+        description = getString(document, "e2eventdescription");
+    }
+
+    private void processAllServicesResult(Document document) {
+        NodeList bouquetList = document.getElementsByTagName("e2bouquet");
+        channels.clear();
+        for(int i = 0; i < bouquetList.getLength(); i++) {
+            Element bouquet = (Element)bouquetList.item(i);
+            NodeList serviceList = bouquet.getElementsByTagName("e2service");
+            for(int j = 0; j < serviceList.getLength(); j++) {
+                Element service = (Element) serviceList.item(j);
+                String id = service.getElementsByTagName("e2servicereference").item(0).getTextContent();
+                String name = service.getElementsByTagName("e2servicename").item(0).getTextContent();
+                channels.put(name, id);
+            }
+        }
+    }
+
+    private String getString(Document document, String elementId) {
+        return Optional.ofNullable(document.getElementsByTagName(elementId)).map(nodeList -> nodeList.item(0)).map(Node::getTextContent).map(String::trim).orElse("");
+    }
+
+    private boolean getBoolean(Document document, String elementId) {
+        return Boolean.parseBoolean(getString(document, elementId));
+    }
+
+    private int getInt(Document document, String elementId) {
+        try {
+            return Integer.parseInt(getString(document, elementId));
+        } catch (NumberFormatException e) {
+            return 0;
+        }
+    }
+
+    public int getVolume() {
+        return volume;
+    }
+
+    public boolean isMute() {
+        return mute;
+    }
+
+    public boolean isPower() {
+        return power;
+    }
+
+    public LocalDateTime getLastAnswerTime() {
+        return lastAnswerTime;
+    }
+
+    public String getChannel() {
+        return channel;
+    }
+
+    public String getTitle() {
+        return title;
+    }
+
+    public String getDescription() {
+        return description;
+    }
+
+    public String getAnswer() {
+        return answer;
+    }
+
+    public Collection<String> getChannels() {
+        return channels.keySet();
+    }
+
+    /**
+     * Getter for Test-Injection
+     * @return HttpGet.
+     */
+    Enigma2HttpClient getEnigma2HttpClient() {
+        return new Enigma2HttpClient(timeout);
```

Comment:
It seems wasteful to create a new object each time. In fact per my ealier suggestion you could get rid of this client class entirely and just call a static method directly instead.

Reply:
Fixed, but not with a static method because of testability